### ISSUE TITLE
Closing Docusign document doesn't trigger any callback, that we can keep track

### ISSUE THREAD

**Issue by oguzozcan** [docusign/mobile-android-sdk#48]
Hello,

We are using 'DSCaptiveSigningListener' in order to read the states of the document, if signing started, canceled or if there is an error.

There is also a finish button upon our document, upon clicking to this button docusign redirects us to a webpage that we set it up.
I want to be able to get a callback when signing or document viewing is completed but i couldn't manage to do so. 
I checked all your interfaces and non of them was getting called upon clicking to this button. 

I also try deleting/changing this redirection url but this didn't change the behaviour. So we are still getting redirected to the same url.

How can i get a callback when user closes the document?

**Comment by naveentds** [comment]
Hi 
We will look into reproducing the issue and if it's an issue, we will fix it in SDK v1.9.4 release.
Thanks.

**Comment by oguzozcan** [comment]
Thanks for your quick reply. Also please let me know incase you need more information or incase everything is working good on your side and i am the one making a mistake :)

**Comment by naveentds** [comment]
Hi
We validated this on our end. We are receiving the onSuccess in the Captive Signing callback listener.

When you generate the captive signing url(/accounts/{account}/envelopes/{envelopeId}/views/recipient) using the eSign REST API, make sure that you provide the following parameter in the request body. 
`{
     "returnUrl": "<url where it needs to redirect after signing is completed>"   // Eg: https://www.docusign.com
  }`

Note that when you complete the signing ceremony, you can observe the redirected url in the network tool Eg: Charles proxy looks like "https://www.docusign.com?event=signing_complete".
Then, in the App, you should receive the onSuccess callback in the Captive Signing listener.

**Comment by oguzozcan** [comment]
Ok but i want to redirect the user back to the app not to a webUrl, plus user can also click finish button in the screen with or without signing in which we still need to go back to the app instead of a url. Maybe it could be achieved with a deeplink but it would not be very straightforward

**Comment by naveentds** [comment]
Hi
When you specify the redirectUrl as mentioned previously, DocuSign backend will append the query param event=signing_complete to the url and SDK will parse this event and sends onSuccess in the callback listener to the App. During this time, the signing ceremony is closed and App can handle onSuccess callback.
When you click Finish button in the screen, DocuSign backend will invoke the return url that you configured. And the SDK parses the event=signing_complete in the url and sends onSuccess callback to the App. It won't show redirected url and instead sends onSuccess callback to the App.

**Comment by oguzozcan** [comment]
Hello again to better show the situation let me send you 2 screenshots. In this scenario, signing is already been completed but user wants to view the document again without taking any action, then closes the form.
<img width="252" alt="Screenshot 2024-05-07 at 11 45 43" src="https://github.com/docusign/mobile-android-sdk/assets/6877185/6475c1e9-24d7-4615-9148-ee3d39c03dd7">
<img width="252" alt="Screenshot 2024-05-07 at 11 46 04" src="https://github.com/docusign/mobile-android-sdk/assets/6877185/14d4f104-b2a8-4a57-b20e-318de7abc1e8">

In this scenario no callback is being called and user is navigated to the signing complete url. User can press return button here inside the browser to return back but we want to directly go back to the app. But no callbacks are being triggered.

**Comment by naveentds** [comment]
Hi
Once the signing is completed, you will see Finish button. On tapping finish button, you will receive the callbacks to the App. Can you confirm that you saw Finish button when you complete signing? And can you let me know how you re opened the document to show close button. Did you use the captive signing api again to launch the same url?

**Comment by oguzozcan** [comment]
Yes i can see finish button when we complete signing but how to just read the document without using captiveSigningApi?

the usage is like this and i can't see any method to just launch the form:
```
                  DocuSign
                        .getInstance()
                        .apply {
                            if (BuildConfig.DEBUG) {
                                setMode(DSMode.DEBUG)
                                allowScreenCapture(true)
                            }
                        }
                        .getSigningDelegate()

                        .launchCaptiveSigning(
                            context = context,
                            envelopeId = event.envelopeId,
                            signingURL = event.url,
                            listener = viewModel,
                            recipientId = null,
                        )
```

Btw sorry for my late response.

**Comment by naveentds** [comment]
You can download and read the document during signing using the hamburger menu and select Download document or Download separate PDFs/documents.

**Comment by oguzozcan** [comment]
Ok but what you are suggesting is a change in the whole experience. Isn't it possible to only preview the document using docusign library and still be able to listen events to the things user is doing? Something simple as closing the docusign preview

**Comment by naveentds** [comment]
You should be able to use docusign esign rest api to download the documents. Checkout https://developers.docusign.com/docs/esign-rest-api/how-to/download-envelope-documents/

**Comment by oguzozcan** [comment]
Thank for the information but for us this is a bug because your iOS and Android clients act differently if i am not missing a configuration difference.

On iOS we get DSMSigningCompleted callback upon clicking either finish or close button. They are using presentCaptiveSigning as equivalent to our launchCaptiveSigning api. But for android callback to signingCompleted method happens only after signing the document. Viewing the document and closing it redirects user to the return url for android and for iOS docusign is calling SigningCompleted callback. 


https://github.com/docusign/mobile-android-sdk/assets/6877185/d01eced5-e070-4b7e-a52c-197f37d966ea

If you want i can provide a video from ios too doing the same thing and also the logs we are receiving

**Comment by naveentds** [comment]
Hi
As per your use-case, looks like there are two steps.
First, it looks like you finished signing using captive signing api and you clicked on Finish button, then you received signingCompleted callback. 
Second, you clicked view form as displayed in the video above. It launched the same captive signing url and it displayed the completed document with close button.
Can you confirm if the scenario mentioned above is accurate.
If that is the case, we will fix the issue in next Android SDK release v1.9.5.

**Comment by oguzozcan** [comment]
Yes its accurate, just as you described it

**Comment by oguzozcan** [comment]
Hello, do you maybe know when new version of the sdk will be released including the fix for this issue? Thank you

**Comment by naveentds** [comment]
We are planning to release SDK v1.9.5 with the fix ~ mid July.

**Comment by naveentds** [comment]
Hi
We released SDK v1.9.5 which fixes the issue. Try it out and let us know if the issue is resolved.

**Comment by oguzozcan** [comment]
Hello, yes issue seemed like fixed. Thank you very much. Closing the ticket
